packages/flutter/lib/src/widgets/conduit.dart

```diff
@@ -0,0 +1,375 @@
+    }
+
+    // Insert to the child slot.
+    final _RenderLeaderProxyBox oldRenderObject = renderObject;
+    assert(slot == null);
+    assert(oldRenderObject.debugValidateChild(child));
+    oldRenderObject.child = child;
+    assert(oldRenderObject == renderObject);
+  }
+
+  @override
+  void moveChildRenderObject(RenderObject child, dynamic slot) {
+    assert(false);
+  }
+
+  @override
+  void removeChildRenderObject(RenderObject child) {
+    assert(child != null);
+    assert(
+      renderObject != null,
+      'Trying to remove $child from $this which does not have a renderObject.',
+    );
+    if (child == renderObject.child) {
+      assert(child?.parent != null);
+      final _RenderLeaderProxyBox oldRenderObject = renderObject;
+      assert(slot == null);
+      assert(renderObject.child == child);
+      oldRenderObject.child = null;
+      assert(oldRenderObject == renderObject);
+    } else {
+      final AnchorKey key = _previousWidget?.remoteKey ?? widget?.remoteKey;
+      assert(key != null);
+      key.currentContext.removeChildRenderObject(child);
+    }
+  }
+}
+
+/// A widget keyed by an [AnchorKey] and allows a [Conduit] possesses the same
+/// [AnchorKey] to attach the render object of its `remoteChild` to.
+class RenderConduitAnchorWidget extends LeafRenderObjectWidget {
+  /// Creates a [RenderConduitAnchorWidget] with an [AnchorKey].
+  const RenderConduitAnchorWidget({
+    @required AnchorKey key,
+  }) : assert(key != null),
+       super(key: key);
+
+  @override
+  AnchorKey get key => super.key as AnchorKey;
+
+  @override
+  LeafRenderObjectElement createElement() => _RenderConduitAnchorElement(this);
+
+  @override
+  RenderObject createRenderObject(BuildContext context) => _RenderFollowerProxyBox();
+}
+
+class _RenderConduitAnchorElement extends LeafRenderObjectElement {
+  _RenderConduitAnchorElement(RenderConduitAnchorWidget widget) : super(widget);
+
+  @override
+  RenderConduitAnchorWidget get widget => super.widget as RenderConduitAnchorWidget;
+
+  @override
+  _RenderFollowerProxyBox get renderObject => super.renderObject as _RenderFollowerProxyBox;
+
+  @override
+  void mount(Element parent, dynamic newSlot) {
+    super.mount(parent, newSlot);
+
+    final _PendingRenderObjectManifest pending = widget.key._manifest.content;
+
+    if (pending == null)
+      return;
+
+    renderObject.leader = pending.leaderRenderBox;
+    final RenderBox pendingRenderBox = pending.renderObject as RenderBox;
+    insertChildRenderObject(pendingRenderBox, null);
+    widget.key._manifest.content = null;
+  }
+
+  @override
+  void unmount() {
+    renderObject.leader = null;
+    super.unmount();
+  }
+
+  @override
+  void insertChildRenderObject(RenderBox child, dynamic slot) {
+    assert(slot == null);
+    final RenderProxyBox oldRenderObject = renderObject;
+    assert(oldRenderObject.debugValidateChild(child));
+    renderObject.child = child;
+    assert(renderObject == oldRenderObject);
```

Comment:
same here (with the "oldRenderObject" vs changing the type on the argument)

Reply:
> What happens if the leader is in a LayoutBuilder that doesn't get laid out until after the follower has already been laid out?

Then the follower's layout boundary will be marked dirty again when the `LayoutBuilder` builds I think?